server/src/spacon/components/mqtt/core.clj

```diff
@@ -21,10 +21,31 @@
             [clojure.core.async :as async]
             [spacon.components.http.auth :refer [get-token]]
             [clojure.tools.logging :as log]
-            [spacon.components.queue.protocol :as queue])
+            [spacon.components.queue.protocol :as queue]
+            [clojure.data.json :as json])
   (:import (org.eclipse.paho.client.mqttv3 MqttException)
            (java.net InetAddress)))
 
+(def vcap_mqtt
+  "Map of keys and values from VCAP_SERVICES mqtt environment variable"
+  (or (some-> (System/getenv "VCAP_SERVICES")
+              (json/read-str :key-fn clojure.core/keyword) :p-rabbitmq first :credentials :protocols :mqtt)
+                {:username :username
```

Comment:
So if there isn't a VCAP_SERVICES environment variable, we'll want to set default mqtt options.  Therefore the values in this map should represent that, ex
```
{:host "localhost" :port 1883 :ssl false}
```